src/lib/plugins/Plugin.ts

```diff
@@ -1,12 +1,13 @@
 import type { ClientOptions } from 'discord.js';
 import type { SapphireClient } from '../SapphireClient';
+import type { Awaited } from '../utils/Types';
 import { postInitialization, postLogin, preGenericsInitialization, preInitialization, preLogin } from './symbols';
 
 // eslint-disable-next-line @typescript-eslint/no-extraneous-class
 export abstract class Plugin {
-	public static [preGenericsInitialization]?: (this: SapphireClient, options?: ClientOptions) => void;
-	public static [preInitialization]?: (this: SapphireClient, options?: ClientOptions) => void;
-	public static [postInitialization]?: (this: SapphireClient, options?: ClientOptions) => void;
-	public static [preLogin]?: (this: SapphireClient, options?: ClientOptions) => void;
-	public static [postLogin]?: (this: SapphireClient, options?: ClientOptions) => void;
+	public static [preGenericsInitialization]?: (this: SapphireClient, options: ClientOptions) => void;
+	public static [preInitialization]?: (this: SapphireClient, options: ClientOptions) => void;
+	public static [postInitialization]?: (this: SapphireClient, options: ClientOptions) => void;
+	public static [preLogin]?: (this: SapphireClient, options: ClientOptions) => Awaited<void>;
+	public static [postLogin]?: (this: SapphireClient, options: ClientOptions) => Awaited<void>;
```

Comment:
why is options no longer optional?

Reply:
Consistency, since Initialization had it first but then login didnt, so I made them all optional but now it came to Login so Initialization also got it. It always gets passed so its not a real issue if its required.

Reply:
ok